### PR TITLE
Extract and convert product tags

### PR DESCRIPTION
**Pull request by zliu1020**
### What it does and why:
Extract and convert product tags

### Checklist:

- [x] Testing & QA: Passes tests and linting.
- [x] :tophat:: I have manually tested these changes.

### Demo:
screen shot 2018-09-26 at 11 12 12 am

---

### Related issues: https://github.com/Shopify/transporter_app/issues/1321
Closes:

### CODE REVIEW

File: lib/shopify_transporter/pipeline/magento/product/top_level_attributes.rb [hunk 1]
```diff
@@ -54,6 +55,12 @@ def image_alt_text(label)
                 label
               end
             end
+
+            def product_tags(input)
+              input['tags'].each_with_object([]) do |tag, array|
```
Comment (JordanVerasamy) [inline lib/shopify_transporter/pipeline/magento/product/top_level_attributes.rb]: Is there a reason we can't just do `input['tags'].map { |tag| tag['name'] }.join(', ')`?

File: lib/shopify_transporter/pipeline/magento/product/top_level_variant_attributes.rb [hunk 2]
```diff
@@ -8,6 +8,7 @@ module Magento
       module Product
         class TopLevelVariantAttributes < Pipeline::Stage
           def convert(hash, record)
+            return if record['variants'].nil?
```
Comment (JordanVerasamy) [inline lib/shopify_transporter/pipeline/magento/product/top_level_variant_attributes.rb]: why is this needed now but wasn't before?

File: spec/shopify_transporter/exporters/magento/product_exporter_spec.rb [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # frozen_string_literal: true
 require 'shopify_transporter/pipeline/stage'
-
+require 'pry'
```
Comment (JordanVerasamy) [inline spec/shopify_transporter/exporters/magento/product_exporter_spec.rb]: remove

**Comment by JordanVerasamy** [comment]
i will also mention that i 🎩 ed this and it works great!

File: lib/shopify_transporter/pipeline/magento/product/top_level_attributes.rb (hunk 1 above)
Comment (JordanVerasamy) [reply lib/shopify_transporter/pipeline/magento/product/top_level_attributes.rb]: you also could do something fancy with inject/reduce, but doing a map then a join is prob the most readable approach

Comment (zliu1020) [reply lib/shopify_transporter/pipeline/magento/product/top_level_attributes.rb]: I used ```each_with_object``` for no particular reason, it was just on the top of my head and ```map``` didn't come to my mind. You're absolutely right and ```map``` seems to be more readable than ```each_with_object```

File: lib/shopify_transporter/pipeline/magento/product/top_level_variant_attributes.rb (hunk 2 above)
Comment (fahia) [reply lib/shopify_transporter/pipeline/magento/product/top_level_variant_attributes.rb]: Probably fails if anything called `simple_product_in_magento_format` as that assumes the presence of `variants`.

File: lib/shopify_transporter/pipeline/magento/product/top_level_attributes.rb (hunk 1 above)
Comment (surajreddy) [reply lib/shopify_transporter/pipeline/magento/product/top_level_attributes.rb]: 👍 to Jordan's recommendation.

File: lib/shopify_transporter/pipeline/magento/product/top_level_variant_attributes.rb (hunk 2 above)
Comment (surajreddy) [inline lib/shopify_transporter/pipeline/magento/product/top_level_variant_attributes.rb]: I think we should remove this return: `return if record['variants'].nil?`

If the variants array doesn't exist, then I suggest that we should set it to an empty array in the Product Record Builder/handle it in the Record Builder. We can also handle it in this file potentially, but I suggest tackling this in a different PR.

For this PR, I suggest removing this line.

Comment (zliu1020) [reply lib/shopify_transporter/pipeline/magento/product/top_level_variant_attributes.rb]: This line of code should've been added when I was working on the ticket about top level variant fields. Converting top level variants makes no sense if there're no variants in the first place. I added this line just for safety.

**Comment by zliu1020** [comment]
I've made changes so that the check against if the input product has a parent is the first thing to be carried out in top level variant conversion. Previously this check won't happen until the accumulation action is about to start. Also as pointed out by @surajreddy , the ProductRecordBuilder class now initializes the parent record with an empty variant array. In this way the downstream code which uses the ProductRecordBuilder doesn't need to bother checking against the existence of variants array.
